api_loader.go

```diff
@@ -585,6 +585,15 @@ func loadApps(specs []*APISpec, muxer *mux.Router) {
 	for i, spec := range specs {
 		go func(spec *APISpec, i int) {
 			subrouter := hostRouters[spec.Domain]
+			if subrouter == nil && spec.Domain != "" {
```

Comment:
I would make this be just `if subrouter == nil {` - otherwise, we could be missing yet another `subrouter == nil` case.

And if we did forget about a separate case, I'd rather have an incorrect warning logged than a crash.